web_console/api/job.js

```diff
@@ -13,54 +21,200 @@ const namespace = config.NAMESPACE;
 const k8s = new KubernetesClient();
 const es = new ElasticSearchClient();
 
+let es_oparator_match_phrase;
+try {
+  es_oparator_match_phrase = require('../es.match_phrase');
+} catch (err) { /* */ }
+
 router.get('/api/v1/jobs', SessionMiddleware, async (ctx) => {
+  const jobs = await Job.findAll({
+    order: [['created_at', 'DESC']],
+  });
   const { flapps } = await k8s.getFLAppsByNamespace(namespace);
-  ctx.body = { data: flapps };
+  const data = jobs.map((job) => ({
+    ...(flapps.items.find((item) => item.metadata.name === job.k8s_name)),
+    localdata: job,
+  }));
+  ctx.body = { data };
 });
 
 router.get('/api/v1/job/:name', SessionMiddleware, async (ctx) => {
   const { name } = ctx.params;
-  const { flapp } = await k8s.getFLApp(namespace, name);
-  ctx.body = { data: flapp };
+  const job = await Job.findOne({ where: { name } });
+  if (!job) {
+    ctx.status = 404;
+    ctx.body = {
+      error: 'Job not found',
+    };
+    return;
+  }
+  const { flapp } = await k8s.getFLApp(namespace, job.k8s_name);
+  ctx.body = {
+    data: {
+      ...flapp,
+      localdata: job,
+    },
+  };
 });
 
-router.get('/api/v1/job/:name/pods', SessionMiddleware, async (ctx) => {
-  const { name } = ctx.params;
-  const { pods } = await k8s.getFLAppPods(namespace, name);
-  ctx.body = { data: pods };
+router.get('/api/v1/job/:k8s_name/pods', SessionMiddleware, async (ctx) => {
+  const { k8s_name } = ctx.params;
+  const { pods } = await k8s.getFLAppPods(namespace, k8s_name);
+  ctx.body = { data: pods.items };
 });
 
-router.get('/api/v1/job/pod/:name/:container', SessionMiddleware, async (ctx) => {
-  const { name, container } = ctx.params;
+router.get('/api/v1/job/:k8s_name/logs/:start_time', SessionMiddleware, async (ctx) => {
```

Comment:
`/api/v1/job/:k8s_name/log?start_time=xxx` would be better, 'caz `es` query DSL is quite complicated

Reply:
for example: we might wanna indicate `end_time` someday